src/VisualStudio/Core/Def/Implementation/ProjectSystem/Legacy/AbstractLegacyProject.cs

```diff
@@ -107,11 +105,12 @@ internal abstract partial class AbstractLegacyProject : ForegroundThreadAffiniti
             ConnectHierarchyEvents();
             RefreshBinOutputPath();
 
-            // TODO: remove this terrible hack, which is working around shims throwing in not-good ways
+            // TODO: The ctor of ExternalErrorDiagnosticUpdateSource throws when running in tests since UIContextImpl calls:
```

Comment:
> // TODO: The ctor of ExternalErrorDiagnosticUpdateSource throws when running in tests since UIContextImpl calls: [](start = 11, length = 113)

GitHub issue